File: PAP2526/test/tp01/TestMyBool.java
package tp01;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class TestMyBool {
	@Test
	void testMajuscule() {
		assertTrue(MyBool.estMajuscule('A'));
		assertTrue(MyBool.estMajuscule('È'));
		assertFalse(MyBool.estMajuscule('a'));
		assertFalse(MyBool.estMajuscule('è'));
	}

	@Test
	void testNbBitV1() {
		assertEquals(0, MyBool.nbBitsV1(0));
		assertEquals(2, MyBool.nbBitsV1(18));
		assertEquals(32, MyBool.nbBitsV1(-1));
	}

	@Test
	void testNbBitV2() {
		assertEquals(0, MyBool.nbBitsV2(0));
		assertEquals(2, MyBool.nbBitsV2(18));
		assertEquals(32, MyBool.nbBitsV2(-1));
	}

	@Test
	void testNbBitV3() {
		assertEquals(0, MyBool.nbBitsV3(0));
		assertEquals(2, MyBool.nbBitsV3(18));
		assertEquals(32, MyBool.nbBitsV3(-1));
	}
	
	@Test
	void testNbBitV3Bis() {
		assertEquals(0, MyBool.nbBitsV3(0));
		assertEquals(2, MyBool.nbBitsV3(18));
		assertEquals(32, MyBool.nbBitsV3(-1));
	}
	
	@Test
	void testIntToHex() {
		assertEquals("0x0",MyBool.intToHex(0));
		assertEquals("0xf4a",MyBool.intToHex(3914));
		assertEquals("0x80000000",MyBool.intToHex(Integer.MIN_VALUE));
		assertEquals("0xffffffff",MyBool.intToHex(-1));
	}
	
	@Test
	void testIntToHexV2() {
		assertEquals("0x0",MyBool.intToHexV2(0));
		assertEquals("0xf4a",MyBool.intToHexV2(3914));
		assertEquals("0x80000000",MyBool.intToHexV2(Integer.MIN_VALUE));
		assertEquals("0xffffffff",MyBool.intToHexV2(-1));
	}
}

File: PAP2526/python/test_my_bool.py
import unittest
import my_bool


class TestMyBool(unittest.TestCase):
    
    def test_estMajuscule(self):
        self.assertTrue(my_bool.estMajuscule('A'))
        self.assertTrue(my_bool.estMajuscule('É'))
        self.assertFalse(my_bool.estMajuscule('a'))
        self.assertFalse(my_bool.estMajuscule('é'))
        
    def test_intToHex(self):
        self.assertEqual("0x0", my_bool.intToHex(0))
        self.assertEqual("0x9", my_bool.intToHex(9))
        self.assertEqual("0xffffffff", my_bool.intToHex(0xffffffff))
        self.assertEqual("0xf4a", my_bool.intToHex(3914))
        
             
if __name__ == "__main__":
    unittest.main()